packages/styled-components/src/sheet/GroupedTag.js

```diff
@@ -36,7 +37,14 @@ class DefaultGroupedTag implements GroupedTag {
     if (group >= this.groupSizes.length) {
       const oldBuffer = this.groupSizes;
       const oldSize = oldBuffer.length;
-      const newSize = BASE_SIZE << ((group / BASE_SIZE) | 0);
+
+      let newSize = oldSize;
+      while (group >= newSize) {
+        newSize <<= 1;
+        if (newSize < 0) {
```

Comment:
This works because eventually we'll hit `1 << 31` the upper limit of SMIs, where we overflow into the signed bit, which turns `newSize` negative.